### PR TITLE
AG-8564 Axis animation

### PR DESCRIPTION
**Pull request by lsjroberts**
PR for https://ag-grid.atlassian.net/browse/AG-8564

Note: We should remove the grid lines on the secondary axes as they look confusing in animation and cause some rendering errors when drawing over other grid lines, but I'll look into that in a future PR.

https://github.com/ag-grid/ag-grid/assets/3817697/6575778a-a077-46a3-9bb9-044b2f52bb44

### CODE REVIEW

File: charts-community-modules/ag-charts-community/src/motion/states.ts [hunk 1]
```diff
@@ -14,24 +15,24 @@ interface StateDefinition<State extends string, Event extends string> {
 }
 
 export class StateMachine<State extends string, Event extends string> {
+    static DEBUG = () => [true, 'animation'].includes(windowValue('agChartsDebug') as string) ?? false;
```
Comment (alantreadway) [inline charts-community-modules/ag-charts-community/src/motion/states.ts:18]: Nice :+1:

**Review by alantreadway — APPROVED** [review]
LGTM

**Review by alexanderby — APPROVED** [review]
I checked out the branch and checked that polar axes don't break. They didn't animate, but hopefully this should be not too difficult.

**Comment by lsjroberts** [comment]
- [ ] Fix snapshots

**Pull request merged**